### PR TITLE
[#140237] Strong params part 1

### PR DESCRIPTION
**Pull request by meara**

### CODE REVIEW

File: app/controllers/affiliates_controller.rb [hunk 1]
```diff
@@ -5,14 +5,12 @@ def new
   end
 
   def create
-    attrs = params[:affiliate]
-
-    if attrs.blank?
+    if affiliate_params.blank?
```
Comment (jhanggi) [inline app/controllers/affiliates_controller.rb]: Do we think this is a legitimate code path? Could we get away with relying on the fact the `Affiliate` will be invalid if we're missing the params?

File: app/controllers/affiliates_controller.rb [hunk 2]
```diff
@@ -38,14 +36,12 @@ def update
     rescue ActiveRecord::RecordNotFound
       flash[:error] = "Affiliate with id #{id} not found!"
     else
-      attrs = params[:affiliate]
-
-      if attrs.blank?
+      if affiliate_params.blank?
```
Comment (jhanggi) [inline app/controllers/affiliates_controller.rb]: Same here

File: app/controllers/file_uploads_controller.rb [hunk 3]
```diff
@@ -37,7 +37,7 @@ def download
 
   # POST /facilities/1/services/3/files
   def create
-    @file = @product.stored_files.new(params[:stored_file].merge(created_by: session_user.id))
+    @file = @product.stored_files.new(create_params.merge(created_by: session_user.id))
```
Comment (jhanggi) [inline app/controllers/file_uploads_controller.rb]: Do we have a spec around this? I'm only about 50% confident that `merge` will keep `created_by` allowed.

File: app/controllers/product_access_groups_controller.rb [hunk 4]
```diff
@@ -50,4 +50,14 @@ def destroy
     end
   end
 
+  private
+
+  def create_params
```
Comment (jhanggi) [inline app/controllers/product_access_groups_controller.rb]: Since create/update are identical, should they be just one method?

Comment (meara) [reply app/controllers/product_access_groups_controller.rb]: cancancan looks for the methods to match the controller action names, so I think we need both.

File: app/controllers/file_uploads_controller.rb (hunk 3 above)
Comment (meara) [reply app/controllers/file_uploads_controller.rb]: Added a spec. Looks like the merge is fine.

**Pull request merged**